zeppelin-zengine/src/main/java/org/apache/zeppelin/scheduler/RemoteScheduler.java

```diff
@@ -34,14 +36,14 @@
 public class RemoteScheduler extends AbstractScheduler {
   private static final Logger LOGGER = LoggerFactory.getLogger(RemoteScheduler.class);
 
-  private RemoteInterpreter remoteInterpreter;
-  private ExecutorService executor;
+  private final RemoteInterpreter remoteInterpreter;
+  private final ExecutorService executor;
 
   public RemoteScheduler(String name,
-                         ExecutorService executor,
                          RemoteInterpreter remoteInterpreter) {
     super(name);
-    this.executor = executor;
+    this.executor =
```

Comment:
The RemoteScheduler creates its own executor. There is now one FIFO executor per RemoteScheduler.